### ISSUE TITLE
Gql gives 500 Internal Error vs. expected GraphQL explain on mis-spelled fragments

### ISSUE THREAD

**Issue by narration-sd** [craftcms/cms#5016]
### Description

A mis-formed or mis-spelled inline fragment signature gets the 500, where usually an informative GraphQL error would have been presented.

### Steps to reproduce

0. GraphQL Explore works fine for this (expand its name to Explore (GraphiQL) for simple cross-GraphQL-platform clarity??)
1. try a simple standard Gql query that has an inline fragment, as most will:
```
query  {
        entries {
          id
          ...on cards_cards_Entry {
            title
          }
        }
      }
```
2. Works
3. now try mis-spelling the signature of the fragment even slightly -- or make it named as for CraftQL, or prefixed as for Gridsome's federativity, or Gatsby's apparent lean in that direction...) -- but we'll be  simple here:
```
query  {
        entries {
          id
          ...on cards_cards_Entryy {
            title
          }
        }
      }
```
4. Get a 500 Internal Server Error, with the query error  'Something went wrong when processing the GraphQL query.' message and a big stack trace. 

### Additional info

PHP version	7.1.28
OS version	Linux 4.15.0-47-generic
Database driver & version	MySQL 5.7.25
Image driver & version	GD 7.1.28
Craft edition & version	Craft Pro 3.3.3
Yii version	2.0.21
Twig version	2.11.3
Guzzle version	6.3.3
Imagine version	1.2.3-dev

Plugins
CraftQL	1.3.4
Element API	2.6.0
Live Vue	dev-craft-gql-field-args
Redactor	2.4.0

**Comment by andris-sevcenko** [comment]
Ah. This is a victim of the rather aggressive performance optimizations, where the entire schema is not loaded for queries that are not introspection. I'll see what I can do to alleviate this.

**Comment by narration-sd** [comment]
nice to keep any performance enhancements -- maybe  you can intercept and go back if the 500 is trying  to appear, something like that? I am sure you probably thought of this already....!

**Comment by andris-sevcenko** [comment]
It's not a 500 anymore as of Craft 3.3.6.

It's not the verbose explanation that GraphQL provides, however, because of the schema loading optimizations. Yet.

**Comment by narration-sd** [comment]
moving fast - I'll update...enjoying the 'Yet' 🐟

**Comment by andris-sevcenko** [comment]
This will be fixed in a nice manner for 3.4.

**Comment by andris-sevcenko** [comment]
Okay, so, if devmode enabled, you'll see the nice verbose GraphQL error messages in 3.4.

You can also customize this behavior using the appropriate event, as of 5fcc5e1a76685a5c02414b752ded07c7ac953910

**Comment by andris-sevcenko** [comment]
This is now available as of Craft 3.4 beta.
